### PR TITLE
Sync `connectionTimeout` and `readTimeout` to Android

### LINKED ISSUE #1149: How to set SocketTimeout longer
Caused by: java.net.SocketTimeoutException: failed to connect to XXXX/XX.XX.130.150 (port 443) from /XX.XX.3.122 (port 41780) after 5000ms

how can I set SocketTimeout longger

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Sync `connectionTimeout` and `readTimeout` to Android

## :bulb: Motivation and Context

Closes #1149

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/lib/src/native_scope_observer.dart [hunk 1]
```diff
@@ -46,7 +46,7 @@ class NativeScopeObserver implements ScopeObserver {
 
   @override
   Future<void> setTag(String key, String value) async {
-    await _sentryNative.setExtra(key, value);
+    await _sentryNative.setTag(key, value);
```
Comment (marandaneto) [inline flutter/lib/src/native_scope_observer.dart:49]: Ups, can we add a test for that?

File: flutter/lib/src/sentry_flutter_options.dart [hunk 2]
```diff
@@ -199,6 +199,12 @@ class SentryFlutterOptions extends SentryOptions {
   /// Only available on iOS and macOS.
   bool enableAppHangTracking = true;
 
+  /// Connection timeout. This will only be synced to the Android native SDK.
+  Duration connectionTimeout = Duration(seconds: 5);
+
+  /// Read timeout. This will only be synced to the Android native SDK.
+  Duration readTimeout = Duration(seconds: 5);
```
Comment (marandaneto) [inline flutter/lib/src/sentry_flutter_options.dart:206]: Are the default 5s for both of them on Android, right?

**Comment by krystofwoldrich** [comment]
I haven't seen any issue about this in RN, but could be potentially also useful I think.

File: flutter/lib/src/sentry_flutter_options.dart (hunk 2 above)
Comment (denrase) [reply flutter/lib/src/sentry_flutter_options.dart:206]: Yeah, those values are taken form `sentry-java`. https://github.com/getsentry/sentry-java/blob/8ec2c16fc2919a60e1ac83845da19fdce6f71219/sentry/src/main/java/io/sentry/SentryOptions.java#L261

**Comment by marandaneto** [comment]
> I haven't seen any issue about this in RN, but could be potentially also useful I think.

Create an issue, but low priority if not asked yet.

**Comment by marandaneto** [comment]
@denrase there are a few linting issues.

**Comment by denrase** [comment]
@marandaneto Ah, i see you you fixed the linting issues already.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**